tests/transforms/test_split_non_commuting.py

```diff
@@ -122,765 +119,580 @@ def complex_wires_processing_fn(results):
+        result = circuit(params)
 
-        dev = qml.device("default.qubit", wires=2)
-        H = qml.Hamiltonian([1.5, 2.5], [qml.I(), qml.I()])
-
-        @partial(split_non_commuting, grouping_strategy=grouping_strategy)
-        @qml.qnode(dev)
-        def circuit():
-            return qml.expval(H)
-
-        with dev.tracker:
-            res = circuit()
-        assert dev.tracker.totals == {}
-        assert qml.math.allclose(res, 4.0)
-
-    @pytest.mark.parametrize("grouping_strategy", [None, "default", "qwc", "wires"])
-    def test_no_obs_tape(self, grouping_strategy):
-        """Tests tapes with only constant offsets (only measurements on Identity)"""
-
-        _dev = qml.device("default.qubit", wires=1)
-
-        @qml.qnode(_dev)
-        def circuit():
-            return qml.expval(1.5 * qml.I(0))
-
-        circuit = split_non_commuting(circuit, grouping_strategy=grouping_strategy)
-
-        with _dev.tracker:
-            res = circuit()
-
-        assert _dev.tracker.totals == {}
-        assert qml.math.allclose(res, 1.5)
+        def _recursively_check_shape(_result, _expected_shape):
+            """Recursively check the shape of _result and _expected_shape.
 
-    @pytest.mark.parametrize("grouping_strategy", [None, "default", "qwc", "wires"])
-    def test_no_obs_tape_multi_measurement(self, grouping_strategy):
-        """Tests tapes with only constant offsets (only measurements on Identity)"""
+            ``qml.math.shape`` will not work in this case because there are arrays of different
+            shapes and dictionaries nested in the results.
 
-        _dev = qml.device("default.qubit", wires=1)
+            """
 
-        @qml.qnode(_dev)
-        def circuit():
-            return qml.expval(1.5 * qml.I()), qml.expval(2.5 * qml.I())
-
-        circuit = split_non_commuting(circuit, grouping_strategy=grouping_strategy)
+            if not _expected_shape:
+                return
 
-        with _dev.tracker:
-            res = circuit()
+            assert len(_result) == _expected_shape[0]
+            for res in _result:
+                _recursively_check_shape(res, _expected_shape[1:])
 
-        assert _dev.tracker.totals == {}
-        assert qml.math.allclose(res, [1.5, 2.5])
+        _recursively_check_shape(result, expected_shape)
 
-    def test_non_pauli_obs_in_circuit(self):
-        """Tests that the tape is executed correctly with non-pauli observables"""
+    @pytest.mark.parametrize("grouping_strategy", [None, "qwc", "wires"])
+    @pytest.mark.parametrize(
+        "obs, expected_result",
+        [
+            ([qml.Hamiltonian([1.5, 2.5], [qml.I(0), qml.I(1)])], 4.0),
+            ([1.5 * qml.I(), 2.5 * qml.I()], [1.5, 2.5]),
+        ],
+    )
+    def test_only_constant_offset(self, grouping_strategy, obs, expected_result):
+        """Tests that split_non_commuting can handle a circuit only measuring Identity."""
```

Comment:
Do we need an execution for this test?

Reply:
Yes, because this is a special case that resulted in no execution at all (zero tapes), and we want to make sure that it does not error out or something.